editor.planx.uk/src/@planx/components/shared/Preview/Card.tsx

```diff
@@ -48,12 +49,23 @@ const Card: React.FC<Props> = ({
   ...props
 }) => {
   const theme = useTheme();
-  const [path, visibleNode] = useStore((state) => [
+  const [path, visibleNode, breadcrumbs, flow] = useStore((state) => [
     state.path,
     state.currentCard,
+    state.breadcrumbs,
+    state.flow,
   ]);
+
+  // Check if we have a Send node in our breadcrumbs
+  //   This is a better/more immediate proxy for "Submitted" because actual send events that populate lowcal_sessions.submitted_at are queued via Hasura
+  const hasSent = Object.keys(breadcrumbs)
+    .reverse()
```

Comment:
nit: As breadcrumb keys are sorted alphabetically not chronologically this doesn't offer any benefits (despite being a clever and nice optimisation!)

Reply:
ah shoot yes always forget that alphabetical key storage ! I'll remove the `.reverse()` here and just call `.some()` directly (at least it stops at first find)

Very much assuming it's NOT worth the expense of sorting, then reversing, then calling `.some()` - but maybe one to revisit in future